collectors/python.d.plugin/edgex/edgex.chart.py

```diff
@@ -0,0 +1,267 @@
+            queue.put(dict())
+
+    return w
+
+class Service(UrlService):
+    def __init__(self, configuration=None, name=None):
+        UrlService.__init__(self, configuration=configuration, name=name)
+        self.order = ORDER
+        self.definitions = CHARTS
+        self.url = '{protocol}://{host}'.format(
+            protocol=self.configuration.get('protocol', 'http'),
+            host=self.configuration.get('host', 'localhost')
+        )
+        self.edgex_ports = {
+            'support_logging': self.configuration.get('port_logging', '48061'),
+#            'sys_mgmt': self.configuration.get('port_sys_mgmt', '48090'),
+            'core_data': self.configuration.get('port_data', '48080'),
+            'core_metadata': self.configuration.get('port_metadata', '48081'),
+            'core_command': self.configuration.get('port_command', '48082')
+        }
+        self.url_core_data = '{url}:{port}/api/v1/'.format(
+            url = self.url,
+            port = self.edgex_ports['core_data']
+        )
+        self.url_core_metadata = '{url}:{port}/api/v1/'.format(
+            url = self.url,
+            port = self.edgex_ports['core_metadata']
+        )
+        self.url_core_command = '{url}:{port}/api/v1/'.format(
+            url = self.url,
+            port = self.edgex_ports['core_command']
+        )
+        self.url_support_logging = '{url}:{port}/api/v1/'.format(
+            url = self.url,
+            port = self.edgex_ports['support_logging']
+        )
+
+
+    def check(self):
+        self.methods = [
+            METHODS(
+                get_data=self._get_core_throughput_events,
+                url=self.url_core_data,
+                run=self.configuration.get('events_per_second', True),
+            ),
+            METHODS(
+                get_data=self._get_core_throughput_readings,
+                url=self.url_core_data,
+                run=self.configuration.get('events_per_second', True)
+            ),
+            METHODS(
+                get_data=self._get_device_info,
+                url=self.url_core_metadata,
+                run=self.configuration.get('number_of_devices', True)
+            ),
+            METHODS(
+                get_data=self._get_metrics_data,
+                url=self.url_core_metadata,
+                run=self.configuration.get('metrics', True)
+            ),
+            METHODS(
+                get_data=self._get_metrics_data,
+                url=self.url_core_data,
+                run=self.configuration.get('metrics', True)
+            ),
+            METHODS(
+                get_data=self._get_metrics_data,
+                url=self.url_core_command,
+                run=self.configuration.get('metrics', True)
+            ),
+            METHODS(
+                get_data=self._get_metrics_data,
+                url=self.url_support_logging,
+                run=self.configuration.get('metrics', True)
+            )
+         ]
```

Comment:
I think we could move initing methods to `init_methods` and invoke it during init and delete `check` method